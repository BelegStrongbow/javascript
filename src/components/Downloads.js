import React from "react";
import styled from "styled-components";
import colors from "yoast-components/style-guide/colors.json";
import Download from "./Download";

const DownloadsContainer = styled.div`
	display: flex;
	background-color:  ${ colors.$color_white };
	margin: 20px 0 0;
	padding: 24px;
	flex-wrap: wrap;
	align-items: flex-start;
`;

/**
 * Returns the rendered Downloads Page component.
 *
 * @param {Object} props The props to use.
 *
 * @returns {ReactElement} The rendered download page.
 */
export default class Downloads extends React.Component {
	render() {
		let props = this.props;
		return (
				<DownloadsContainer>
					{ props.downloads.map( function( download ) {
						return <Download
							key={ download.id }
							name={ download.name }
							currentVersion={ download.currentVersion }
							icon={ download.icon }
							buttons={ download.buttons }
						/>;
					} ) }
				</DownloadsContainer>
		);
	}
}
