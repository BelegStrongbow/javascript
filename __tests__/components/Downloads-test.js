import React from 'react';
import { createComponentWithIntl } from "../../utils";
import Downloads from '../../src/components/Downloads';
import SeoIcon from "../../src/icons/Yoast/Yoast_SEO_Icon_Small.svg";

test('The Downloads component matches the snapshot', () => {
	let downloads = [
		{
			product: "Yoast SEO",
			version: "version 4.7",
			iconSource: SeoIcon,
			buttons: [
				{ label: "for Wordpress",
					onButtonClick: () => {} },
				{ label: "for Drupal",
					onButtonClick: () => {} },
			],
		}, {
			product: "Local SEO",
			version: "version 4.7",
			iconSource: SeoIcon,
			buttons: [
				{ label: "for Wordpress",
					onButtonClick: () => {} },
			],
		}, {
			product: "Video SEO",
			version: "version 4.7",
			iconSource: SeoIcon,
			buttons: [
				{ label: "for Wordpress",
					onButtonClick: () => {} },
				{ label: "for Drupal",
					onButtonClick: () => {} },
				{ label: "for Typo3",
					onButtonClick: () => {} },
			],
		}, {
			product: "News SEO",
			version: "version 4.7",
			iconSource: SeoIcon,
			buttons: [
				{ label: "for Wordpress",
					onButtonClick: () => {} },
				{ label: "for Drupal",
					onButtonClick: () => {} },
				{ label: "for Typo3",
					onButtonClick: () => {} },
			],
		}, {
			product: "Yoast SEO for WooCommerce",
			version: "version 4.7",
			iconSource: SeoIcon,
			buttons: [
				{ label: "for Wordpress",
					onButtonClick: () => {} },
			],
		},
	];


	const component = createComponentWithIntl(
		<Downloads downloads={ downloads }/>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
