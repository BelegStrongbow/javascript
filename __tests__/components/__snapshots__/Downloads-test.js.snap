exports[`test The Downloads component matches the snapshot 1`] = `
<div
  className="bFpGra">
  <div
    className="ijckaj">
    <h2
      className="eUCYpf">
      Yoast SEO
    </h2>
    <p
      className="hTJUTU">
       
      version 4.7
    </p>
    <img
      alt=""
      className="ieMZJu"
      src="Yoast_SEO_Icon_Small.svg" />
    <ul
      className="etVRNn">
      <li
        className="jVUpbY">
         
        <button
          className="gAzDzu"
          onClick={[Function]}>
          Download
        </button>
        <p
          className="bOQKYc">
          for Wordpress
        </p>
      </li>
      <li
        className="jVUpbY">
         
        <button
          className="gAzDzu"
          onClick={[Function]}>
          Download
        </button>
        <p
          className="bOQKYc">
          for Drupal
        </p>
      </li>
    </ul>
  </div>
  <div
    className="ijckaj">
    <h2
      className="eUCYpf">
      Local SEO
    </h2>
    <p
      className="hTJUTU">
       
      version 4.7
    </p>
    <img
      alt=""
      className="ieMZJu"
      src="Yoast_SEO_Icon_Small.svg" />
    <ul
      className="etVRNn">
      <li
        className="jVUpbY">
         
        <button
          className="gAzDzu"
          onClick={[Function]}>
          Download
        </button>
        <p
          className="bOQKYc">
          for Wordpress
        </p>
      </li>
    </ul>
  </div>
  <div
    className="ijckaj">
    <h2
      className="eUCYpf">
      Video SEO
    </h2>
    <p
      className="hTJUTU">
       
      version 4.7
    </p>
    <img
      alt=""
      className="ieMZJu"
      src="Yoast_SEO_Icon_Small.svg" />
    <ul
      className="etVRNn">
      <li
        className="jVUpbY">
         
        <button
          className="gAzDzu"
          onClick={[Function]}>
          Download
        </button>
        <p
          className="bOQKYc">
          for Wordpress
        </p>
      </li>
      <li
        className="jVUpbY">
         
        <button
          className="gAzDzu"
          onClick={[Function]}>
          Download
        </button>
        <p
          className="bOQKYc">
          for Drupal
        </p>
      </li>
      <li
        className="jVUpbY">
         
        <button
          className="gAzDzu"
          onClick={[Function]}>
          Download
        </button>
        <p
          className="bOQKYc">
          for Typo3
        </p>
      </li>
    </ul>
  </div>
  <div
    className="ijckaj">
    <h2
      className="eUCYpf">
      News SEO
    </h2>
    <p
      className="hTJUTU">
       
      version 4.7
    </p>
    <img
      alt=""
      className="ieMZJu"
      src="Yoast_SEO_Icon_Small.svg" />
    <ul
      className="etVRNn">
      <li
        className="jVUpbY">
         
        <button
          className="gAzDzu"
          onClick={[Function]}>
          Download
        </button>
        <p
          className="bOQKYc">
          for Wordpress
        </p>
      </li>
      <li
        className="jVUpbY">
         
        <button
          className="gAzDzu"
          onClick={[Function]}>
          Download
        </button>
        <p
          className="bOQKYc">
          for Drupal
        </p>
      </li>
      <li
        className="jVUpbY">
         
        <button
          className="gAzDzu"
          onClick={[Function]}>
          Download
        </button>
        <p
          className="bOQKYc">
          for Typo3
        </p>
      </li>
    </ul>
  </div>
  <div
    className="ijckaj">
    <h2
      className="eUCYpf">
      Yoast SEO for WooCommerce
    </h2>
    <p
      className="hTJUTU">
       
      version 4.7
    </p>
    <img
      alt=""
      className="ieMZJu"
      src="Yoast_SEO_Icon_Small.svg" />
    <ul
      className="etVRNn">
      <li
        className="jVUpbY">
         
        <button
          className="gAzDzu"
          onClick={[Function]}>
          Download
        </button>
        <p
          className="bOQKYc">
          for Wordpress
        </p>
      </li>
    </ul>
  </div>
</div>
`;
